addons/crm_iap_lead_enrich/models/crm_lead.py

```diff
@@ -123,21 +122,11 @@ def _iap_enrich_get_message_data(self, company_data):
             })
         return log_data
 
-    @api.model
-    def _iap_enrich_notify_no_more_credit(self):
-        """ Notify when user has no credits anymore. In order to avoid to spam
-        people each hour, an ir.config_parameter is set. """
-        already_notified = self.env['ir.config_parameter'].sudo().get_param('crm_iap_lead_enrich.credit_notification')
-        if already_notified == 'True':
-            return
-
-        iap_account = self.env['iap.account'].search([('service_name', '=', 'reveal')], limit=1)
-        if not iap_account:
-            return
-
-        mail_template = self.env.ref('crm_iap_lead_enrich.mail_template_data_iap_lead_enrich_nocredit')
-        if not mail_template:
-            return
-
-        mail_template.sudo().send_mail(iap_account.id, force_send=False, notif_layout='mail.mail_notification_light')
-        self.env['ir.config_parameter'].sudo().set_param('lead_enrich.already_notified', True)
+    @api.depends('email_from', 'probability', 'iap_enrich_done', 'reveal_id')
+    def _compute_show_enrich_button(self):
+        for lead in self:
+            config = self.env['ir.config_parameter'].sudo().get_param('crm.lead_enrichment_auto','auto')
+            if lead.active == False or config != 'on_demand' or lead.email_from == False or lead.iap_enrich_done or lead.reveal_id != False or lead.probability in (0, 100):
```

Comment:
Actually no need to loop on leads if config is not on_demand -> if not config or config != 'on_demande': directly set to False

Reply:
actually no need to loop as all as there is only one lead

Reply:
Well compute methods should always work in batch, as it depends on how you modify (triggers will define the set of records to recompute) ... consider self as a true record set with potentially millions of records ^^ .